### PR TITLE
chore: remove Python 3.7 support

### PR DESCRIPTION
**Pull request by Saransh-cpp**
Following https://github.com/scikit-hep/vector/pull/348#issuecomment-1557596467, this PR removes the support of Python 3.7 from vector. I'm still not completely sure if there was a concensus regarding this but I'll be happy to close this PR if we do want to support Python 3.7.

## Description

**Kindly take a look at [CONTRIBUTING.md](https://github.com/scikit-hep/vector/blob/main/.github/CONTRIBUTING.md).**

_Please describe the purpose of this pull request. Reference and link to any relevant issues or pull requests._

## Checklist

- [ ] Have you followed the guidelines in our Contributing document?
- [ ] Have you checked to ensure there aren't any other open Pull Requests for the required change?
- [ ] Does your submission pass pre-commit? (`$ pre-commit run --all-files` or `$ nox -s lint`)
- [ ] Does your submission pass tests? (`$ pytest` or `$ nox -s tests`)
- [ ] Does the documentation build with your changes? (`$ cd docs; make clean; make html` or `$ nox -s docs`)
- [ ] Does your submission pass the doctests? (`$ xdoctest ./src/vector` or `$ nox -s doctests`)

## Before Merging

- [ ] Summarize the commit messages into a brief review of the Pull request.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,7 +17,7 @@
 [![LICENSE][license-badge]][license-link]
 [![Scikit-HEP][sk-badge]][sk-link]
 
-Vector is a Python 3.7+ library for 2D, 3D, and [Lorentz vectors](https://en.wikipedia.org/wiki/Special_relativity#Physics_in_spacetime), especially _arrays of vectors_, to solve common physics problems in a NumPy-like way.
+Vector is a Python 3.8+ library (Python 3.6 and 3.7 supported till `v0.9.0` and `v1.0.0`, respectively) library for 2D, 3D, and [Lorentz vectors](https://en.wikipedia.org/wiki/Special_relativity#Physics_in_spacetime), especially _arrays of vectors_, to solve common physics problems in a NumPy-like way.
```
Comment (Saransh-cpp) [inline README.md:20]:
```suggestion
Vector is a Python 3.8+ (Python 3.6 and 3.7 supported till `v0.9.0` and `v1.0.0`, respectively) library for 2D, 3D, and [Lorentz vectors](https://en.wikipedia.org/wiki/Special_relativity#Physics_in_spacetime), especially _arrays of vectors_, to solve common physics problems in a NumPy-like way.
```

Comment (Saransh-cpp) [inline README.md:20]:
```suggestion
Vector is a Python 3.8+ library (Python 3.6 and 3.7 supported till `v0.9.0` and `v1.0.0`, respectively) for 2D, 3D, and [Lorentz vectors](https://en.wikipedia.org/wiki/Special_relativity#Physics_in_spacetime), especially _arrays of vectors_, to solve common physics problems in a NumPy-like way.
```

**Comment by henryiii** [comment]
Ruff's config still has:
```toml
target-version = "py37"
```

**Comment by Saransh-cpp** [comment]
Oops, fixed it, thank you!

**Comment by henryiii** [comment]
@jpivarski thoughts? Should this go in 1.1? I’m thinking it might be good to do now since 1.1 doesn’t support numba on 3.7, while 1.0 does.

**Comment by jpivarski** [comment]
Yes, it could be good to do all of the dependency version threshold changes in 1.1, instead of using multiple minor versions or using up numbers.

It's a tighter set of version constraints for someone who wants #347, but it doesn't sound like a likely scenario. ("I want `ak.sum` to work on Vectors, but I'm using Python 3.7.")

I'd say go ahead and add it to Vector 1.1.

**Comment by henryiii** [comment]
It's not just about using up numbers, this release is actually a downgrade for Python 3.7 users - they now can't use numba with vector anymore if we don't bump the Python version now. Using `ak.sum` is a new feature and won't break existing code like removing numba could.

**Review by henryiii — APPROVED** [review]

**Pull request merged**